File: hto/06/hto_06_2.c
struct ListNode *reverseList(struct ListNode *node, int *returnSize) {
    if (!node) return NULL;
    struct ListNode *pre = NULL;
    struct ListNode *cur = node;
    while (cur) {
        (*returnSize)++;
        struct ListNode *tmp = cur->next;
        cur->next = pre;
        pre = cur;
        cur = tmp;
    }
    return pre;
}

int *reversePrint(struct ListNode *head, int *returnSize) {
    int *res = (int *)malloc(sizeof(int) * 1024);
    int i = 0;
    *returnSize = 0;
    head = reverseList(head, returnSize);
    while (head) {
        res[i++] = head->val;
        head = head->next;
    }
    return res;
}

File: hto/06/hto_06_2.go
import "fmt"

type ListNode struct {
	Val int 
	Next *ListNode
}

func reversePrint(head *ListNode) []int {
    if head == nil {
        return []int{}
    }
    res := reversePrint(head.Next)
    res = append(res, head.Val)
    return res
}

func main() {
	node1 := &ListNode{1, nil}
	node2 := &ListNode{2, nil}
	node3 := &ListNode{3, nil}
	node1.Next = node2
	node2.Next = node3
	fmt.Println(reversePrint(node1))
}
